specification/cognitiveservices/data-plane/AzureOpenAI/authoring/preview/2025-04-01-preview/azureopenai.json

```diff
@@ -0,0 +1,7164 @@
+          "items": {
+            "$ref": "#/definitions/CreateEvalItem"
+          },
+          "type": "array"
+        },
+        "labels": {
+          "description": "The labels to classify to each item in the evaluation.",
+          "items": {
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "model": {
+          "description": "The model to use for the evaluation. Must support structured outputs.",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the grader.",
+          "type": "string"
+        },
+        "passing_labels": {
+          "description": "The labels that indicate a passing result. Must be a subset of labels.",
+          "items": {
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "type": {
+          "description": "The object type, which is always `label_model`.",
+          "enum": [
+            "label_model"
+          ],
+          "type": "string"
+        }
+      },
+      "required": [
+        "type",
+        "model",
+        "input",
+        "passing_labels",
+        "labels",
+        "name"
+      ],
+      "title": "LabelModelGrader",
+      "type": "object"
+    },
+    "CreateEvalLogsDataSourceConfig": {
+      "description": "A data source config which specifies the metadata property of your stored completions query. This is usually metadata like `usecase=chatbot` or `prompt-version=v2`, etc.",
+      "properties": {
+        "metadata": {
+          "additionalProperties": true,
+          "description": "Metadata filters for the logs data source.",
+          "example": "{\n  \"use_case\": \"customer_support_agent\"\n}\n",
+          "type": "object"
+        },
+        "type": {
+          "default": "logs",
+          "description": "The type of data source. Always `logs`.",
+          "enum": [
+            "logs"
+          ],
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ],
+      "title": "LogsDataSourceConfig",
+      "type": "object"
+    },
+    "CreateEvalRequest": {
+      "properties": {
+        "data_source_config": {
+          "description": "The configuration for the data source used for the evaluation runs.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/CreateEvalCustomDataSourceConfig"
+            },
+            {
+              "$ref": "#/definitions/CreateEvalLogsDataSourceConfig"
+            }
+          ],
+          "type": "object"
+        },
+        "metadata": {
+          "$ref": "#/definitions/Metadata"
+        },
+        "name": {
+          "description": "The name of the evaluation.",
+          "type": "string"
+        },
+        "testing_criteria": {
+          "description": "A list of graders for all eval runs in this group.",
+          "items": {
+            "allOf": [
```

Comment:
This allOf is still describing the sum of all constraints. It would need a polymorphic base type in order to be expressed semantically in the same way as a oneOf in openapi 3.x.

Reply:
We need a base Grader model that is polymorphic, all types in the allOf here need to do an allOf the new grader base model use base model as the items schema here.

Reply:
As we discussed offline, the authoring spec has been changed to openapi 3.0.0.